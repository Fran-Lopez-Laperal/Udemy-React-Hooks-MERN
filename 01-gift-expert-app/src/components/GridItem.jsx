import PropTypes from 'prop-types';

function GridItem({ title, url }) {


  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GridItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired 
}

export default GridItem

