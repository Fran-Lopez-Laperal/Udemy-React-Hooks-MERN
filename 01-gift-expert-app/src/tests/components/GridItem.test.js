import { shallow } from "enzyme";
import GridItem from '../../components/GridItem.jsx'

describe('Pruebas <GridGifItem/> ', () => { 
    const title = 'sdffsdf'
    const url = 'afasd'

    const  wrapper = shallow(<GridItem title={title} url={url}/>)

    test('debe mostrar el componente correctamente',() => {

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de tener un parrafo con el title',()=> {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
        
    })

    test('debe obtener la imagen al url y alt de los props',()=> {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    } )

    test('debe tener animate',()=> {
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true)
    } )




 })