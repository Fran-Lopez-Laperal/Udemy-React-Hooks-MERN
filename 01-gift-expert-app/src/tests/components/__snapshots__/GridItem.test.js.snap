// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GridGifItem/>  debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="sdffsdf"
    src="afasd"
  />
  <p>
    sdffsdf
  </p>
</div>
`;
